web_app/utils/oidc_tools.py

```diff
@@ -0,0 +1,49 @@
+"""Handle user log-in and account info."""
+
+from typing import Any, Dict, Optional, cast
+
+from ..config import cache, oidc
+
+
+class CurrentUser:
+    """Wrap oidc's user info requests."""
+
+    # TODO - cache, clear cache, etc.
+    @staticmethod
+    def _get_info() -> Dict[str, Any]:
+        """Query OIDC."""
+        return cast(
+            Dict[str, Any],
+            oidc.user_getinfo(["preferred_username", "email", "name", "groups"]),
+        )
+
+    @staticmethod
+    def get_summary() -> Optional[Dict[str, Any]]:
+        """Query OIDC."""
+        if not CurrentUser.is_authenticated():
+            return None
+        return CurrentUser._get_info()
+
+    @staticmethod
+    def is_authenticated() -> bool:
+        """Is the user authenticated (logged-in)?"""
+        return bool(oidc.user_loggedin)
```

Comment:
Can just any user in the collab be authenticated?  What does that mean for editing?  (wondering if that gives you power to edit your inst, or where that happens)

Reply:
anyone from an institution can edit their institution's info. If they try to view an institution they don't belong to, they're redirected to their's

Reply:
We could create another group, the downside is that this is prescriptive.

Reply:
ILs may also designate someone to manage MoUs too. This was brought up last time we talked about auth with Paolo

Reply:
Note that right now we have "institution admins" with the `<group_path>/_admin` group.  This could be leveraged here too.

Reply:
I'm not sure if we want to extend permissions *that* far beyond MoU access. If there's a project manager-like role at an institution, I could see them having access to MoUs but nothing beyond that which would be "institution admins"-level

Reply:
We could also do the similar thing the author list is doing, which is to have a subgroup under each institution.  Thus you'd have `<group_path>/mou_admin`.

Reply:
These are of course just options.  The project owner (Paolo?) needs to speak about the requirements here.